src/main/java/teammates/ui/webapi/action/EnrollStudentsAction.java

```diff
@@ -45,148 +41,56 @@ public void checkSpecificAccessControl() {
     }
 
     @Override
-    public ActionResult execute() throws EntityNotFoundException {
+    public JsonResult execute() throws EntityNotFoundException {
 
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-        String studentsInfo = getRequestBody();
-
-        /* Process enrollment list and setup data for page result */
-        try {
-            List<StudentAttributes>[] students = enrollAndProcessResultForDisplay(studentsInfo, courseId);
-            EnrollResults dataFormat = new EnrollResults(getInstructorCourseEnrollResult(students));
-            return new JsonResult(dataFormat);
-
-        } catch (EntityDoesNotExistException e) {
-            throw new EntityNotFoundException(e);
-
-        } catch (EnrollException | InvalidParametersException e) {
-            return new JsonResult(e.getMessage(), HttpStatus.SC_BAD_REQUEST);
-
-        } catch (EntityAlreadyExistsException e) {
-            log.severe("Entity already exists exception occurred when updating student: " + e.getMessage());
-
-            return new JsonResult("The enrollment failed, possibly because some students were re-enrolled before "
-                            + "the previous enrollment action was still being processed by TEAMMATES database "
-                            + "servers. Please try again after about 10 minutes. If the problem persists, "
-                            + "please contact TEAMMATES support", HttpStatus.SC_CONFLICT);
+        if (!logic.isCoursePresent(courseId)) {
```

Comment:
You may not want to include this as `checkSpecificAccessControl` ensure the present of the course.